NitroxClient/MonoBehaviours/Gui/Settings/SettingsOptions.cs

```diff
@@ -0,0 +1,103 @@
+﻿using NitroxClient.Unity.Helper;
+using NitroxModel.Logger;
+using Steamworks;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.UI;
+
+namespace NitroxClient.MonoBehaviours.Gui.Settings {
+    public class SettingsOptions {
+        public void AddInputFieldOption(int tabIndex, string label, string placeholder, string value, uGUI_OptionsPanel uGUI_Options, UnityAction<string> callback = null) {
+            uGUI_MainMenu uGUI_Main = uGUI_MainMenu.main;
+            uGUI_TabbedControlsPanel uGUI_Control = uGUI_Main.GetComponentInChildren<uGUI_TabbedControlsPanel>();
+            GameObject InputOption = new GameObject();
+            GameObject input_field = Object.Instantiate(uGUI_Options.keyRedemptionPrefab.transform.GetChild(1).gameObject);
+            GameObject caption = Object.Instantiate(uGUI_Options.toggleOptionPrefab.transform.GetChild(0).GetChild(2).gameObject);
+
+            caption.GetComponent<Text>().text = label;
+            input_field.transform.SetParent(caption.transform);
+            caption.transform.SetParent(InputOption.transform);
+
+            InputOption.name = "uGUI_InputFieldOption";
+            InputOption.AddComponent<LayoutElement>();
+            InputOption.GetComponent<LayoutElement>().minHeight = 60;
+            InputOption.GetComponent<LayoutElement>().preferredWidth = -1;
+            InputOption.GetComponent<LayoutElement>().preferredHeight = 60;
+            input_field.transform.localPosition = new Vector3(750, 15, input_field.transform.localPosition.z);
+
+            GameObject gameObject = uGUI_Control.AddItem(tabIndex, InputOption);
+
+            InputField componentInChildren = gameObject.GetComponentInChildren<InputField>();
+            Utils.Assert(componentInChildren != null, "see log", null);
```

Comment:
We have our own `Validate.NotNull()`. I wonder though if `"see log"` is applicable to this usecase.